fe/fe-core/src/main/java/com/starrocks/load/DeleteJob.java

```diff
@@ -21,178 +21,164 @@
     }
 
     public MultiDeleteInfo getDeleteInfo() {
         return deleteInfo;
     }
 
-    public String getLabel() {
-        return this.label;
-    }
-
-    public Set<PushTask> getPushTasks() {
-        return pushTasks;
-    }
-
-    @Override
-    public long getId() {
-        return this.id;
-    }
-
     @Override
     public void afterVisible(TransactionState txnState, boolean txnOperated) {
         if (!txnOperated) {
             return;
         }
-        executeFinish();
+        setState(DeleteState.FINISHED);
+        GlobalStateMgr.getCurrentState().getDeleteHandler().recordFinishedJob(this);
+        GlobalStateMgr.getCurrentGlobalTransactionMgr().getCallbackFactory().removeCallback(getId());
         GlobalStateMgr.getCurrentState().getEditLog().logFinishMultiDelete(deleteInfo);
     }
 
     @Override
-    public void afterAborted(TransactionState txnState, boolean txnOperated, String txnStatusChangeReason)
-            throws UserException {
+    public void afterAborted(TransactionState txnState, boolean txnOperated, String txnStatusChangeReason) {
         // just to clean the callback
         GlobalStateMgr.getCurrentGlobalTransactionMgr().getCallbackFactory().removeCallback(getId());
     }
 
-    public void executeFinish() {
-        setState(DeleteState.FINISHED);
-        GlobalStateMgr.getCurrentState().getDeleteHandler().recordFinishedJob(this);
-        GlobalStateMgr.getCurrentGlobalTransactionMgr().getCallbackFactory().removeCallback(getId());
-    }
-
-    public long getTransactionId() {
-        return this.signature;
+    public abstract void run(DeleteStmt stmt, Database db, Table table, List<Partition> partitions)
+            throws DdlException, QueryStateException;
+
+    public abstract long getTimeoutMs();
+
+    public abstract void clear();
+
+    public boolean cancel(DeleteHandler.CancelType cancelType, String reason) {
+        LOG.info("start to cancel delete job, transactionId: {}, cancelType: {}", getTransactionId(),
+                cancelType.name());
+
+        GlobalTransactionMgr globalTransactionMgr = GlobalStateMgr.getCurrentGlobalTransactionMgr();
+        try {
+            globalTransactionMgr.abortTransaction(getDeleteInfo().getDbId(), getTransactionId(), reason);
+        } catch (Exception e) {
+            TransactionState state =
+                    globalTransactionMgr.getTransactionState(getDeleteInfo().getDbId(), getTransactionId());
+            if (state == null) {
+                LOG.warn("cancel delete job failed because txn not found, transactionId: {}", getTransactionId());
+            } else if (state.getTransactionStatus() == TransactionStatus.COMMITTED ||
+                    state.getTransactionStatus() == TransactionStatus.VISIBLE) {
+                LOG.warn("cancel delete job {} failed because it has been committed, transactionId: {}",
+                        getId(), getTransactionId());
+                return false;
+            } else {
```

Comment:
Are these check really necessary? `globalTransactionMgr.abortTransaction()` should provide the detailed error message itself.

Reply:
I add a new `TransactionAlreadyCommitException` and check it here.